### PR TITLE
WS-029: Sightings search bar

### PR DESCRIPTION
**Pull request by chiara-d697**
Fix bug in backend to ensure username filter is now working

### CODE REVIEW

File: backend/Controllers/SightingsController.cs [hunk 1]
```diff
@@ -33,52 +32,62 @@ IUsersRepo usersRepo
...
-                        LatinName = s.Species.LatinName,
-                        PhotoUrl = s.Species.PhotoUrl,
-                        Description = s.Species.Description,
-                        EndangeredStatus = s.Species.EndangeredStatus
-                    },
-                PhotoUrl = s.PhotoUrl,
-                User = new UserResponse
+            var results = _sightingsRepo
+                .GetAllSightings()
+                .Select(s => new ExtendedSightingResponse
```
Comment (TimLeach635) [inline backend/Controllers/SightingsController.cs:43]: This is best moved to a constructor of `ExtendedSightingResponse`, so you can just change this to `.Select(s => new ExtendedSightingResponse(s))`

File: backend/Controllers/SightingsController.cs [hunk 2]
```diff
@@ -33,52 +32,62 @@ IUsersRepo usersRepo
...
-            }).ToList();
+                        : null,
+                    }
+                )
+                .ToList();
+            return results
+                .Where(s => SearchTerm.LocationId == null || s.Location.Id == SearchTerm.LocationId)
+                .Where(s => SpeciesSearchTerm.SpeciesId == null || s.Species.Id == SpeciesSearchTerm.SpeciesId)
+                .Where(s => UserSearchTerm.CreatedByUserId == null || s.User.Id == UserSearchTerm.CreatedByUserId)
+                .ToList();
```
Comment (TimLeach635) [inline backend/Controllers/SightingsController.cs:90]: Yeahhhh 😎

File: frontend/src/clients/apiClients.ts [hunk 3]
```diff
@@ -80,13 +80,24 @@ function getAuthorizationHeader(username: string, password: string) {
...
+  locationId: number,
+  speciesId: number,
+  createdByUserId: number
+): Promise<Array<Sighting>> {
+  const extraQueries = locationId || speciesId || createdByUserId ? `?` : "";
+  const query = locationId ? `locationId=${locationId}&` : "";
+  const query2 = speciesId ? `speciesId=${speciesId}&` : "";
+  const query3 = createdByUserId ? `createdByUserId=${createdByUserId}&` : "";
+  const response = await fetch(
+    `https://localhost:5001/sightings${extraQueries}${query}${query2}${query3}`,
```
Comment (TimLeach635) [inline frontend/src/clients/apiClients.ts:93]: You'll need to merge from `main` and change this to use the `baseUrl` variable

File: frontend/src/clients/apiClients.ts [hunk 4]
```diff
@@ -229,6 +240,10 @@ export const getPopularLocations = async () => {
   return data;
 };
 
+export async function fetchUsers(): Promise<Array<User>> {
+  const response = await fetch(`https://localhost:5001/users`);
```
Comment (TimLeach635) [inline frontend/src/clients/apiClients.ts:244]: ditto

**Review by TimLeach635 — CHANGES_REQUESTED** [review]
A few comments!

File: backend/Controllers/SightingsController.cs (hunk 2 above)
Comment (chiara-d697) [reply backend/Controllers/SightingsController.cs:90]: Wooop boolean algebra

File: frontend/src/clients/apiClients.ts (hunk 3 above)
Comment (chiara-d697) [reply frontend/src/clients/apiClients.ts:93]: Done and changed everywhere

File: frontend/src/clients/apiClients.ts (hunk 4 above)
Comment (chiara-d697) [reply frontend/src/clients/apiClients.ts:244]: Done

File: backend/Controllers/SightingsController.cs (hunk 1 above)
Comment (chiara-d697) [reply backend/Controllers/SightingsController.cs:43]: Done

**Review by TimLeach635 — APPROVED** [review]

**Pull request merged**